### PR TITLE
[onert] Support ODC cross build

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit updates build infra to support ODC cross build.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

### CODE REVIEW

File: Makefile.template [hunk 1]
```diff
@@ -142,20 +150,24 @@ distclean:
 $(WORKSPACE):
 	mkdir -p $@
 
+prepare_buildtool_internal: $(WORKSPACE)
+	cmake -S infra/buildtool -B $(BUILDTOOL_WORKSPACE)/obj -DBUILDTOOL_PATH=$(BUILDTOOL_PATH)
+	cmake --build $(BUILDTOOL_WORKSPACE)/obj/ -j$(NPROCS)
 prepare_nncc_internal: $(WORKSPACE)
-ifneq ($(CROSS_BUILD),1)
-	./nncc configure -DBUILD_GTEST=OFF -DENABLE_TEST=OFF -DEXTERNALS_BUILD_THREADS=$(NPROCS) -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) \
+ifeq (,$(findstring android,$(TARGET_OS)))
```
Comment (jyoungyun) [inline Makefile.template:157]: If `TARGET_OS` contains `android` string, will the following function be executed?

File: infra/nnfw/cmake/packages/HDF5Config.cmake [hunk 2]
```diff
@@ -51,7 +63,7 @@ else()
   endif()
 
   # We can use "hdf5" and "hdf5_cpp" to use the same file founded with above.
-  list(APPEND HDF5_CXX_LIBRARIES "hdf5" "hdf5_cpp" "sz" "z" "dl" "m")
+  list(APPEND HDF5_CXX_LIBRARIES ${HDF5_CXX_LIBRARY_hdf5_cpp} ${HDF5_CXX_LIBRARY_hdf5} "sz" "z" "dl" "m")
```
Comment (jyoungyun) [inline infra/nnfw/cmake/packages/HDF5Config.cmake:66]: (minor comment)
```suggestion
  list(APPEND HDF5_CXX_LIBRARIES ${HDF5_CXX_LIBRARY_hdf5} ${HDF5_CXX_LIBRARY_hdf5_cpp} "sz" "z" "dl" "m")
```

File: Makefile.template (hunk 1 above)
Comment (hseok-oh) [reply Makefile.template:157]: No. If `TARGET_OS` does not contain.

Comment (jyoungyun) [reply Makefile.template:157]: Ah, I got it. This expression is compare '' string with findstring. Thank you :)

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by glistening — APPROVED** [review]
LGTM

**Pull request merged**